cmd/limactl/volume.go

```diff
@@ -0,0 +1,235 @@
+	var volumeCreateCommand = &cobra.Command{
+		Use: "create VOLUME",
+		Example: `
+To create a new volume:
+$ limactl volume create VOLUME
+`,
+		Short: "Create a Lima volume",
+		Args:  cobra.ExactArgs(1),
+		RunE:  volumeCreateAction,
+	}
+	volumeCreateCommand.Flags().String("size", "2G", "configure the volume size")
+	return volumeCreateCommand
+}
+
+func volumeCreateAction(cmd *cobra.Command, args []string) error {
+	_, err := createVolume(cmd, args)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func createVolume(cmd *cobra.Command, args []string) (*store.Volume, error) {
+	// only exactly one arg is allowed (?)
+	name := args[0]
+
+	volDir, err := store.VolumeDir(name)
+	if err != nil {
+		return nil, err
+	}
+
+	if _, err := os.Stat(volDir); !errors.Is(err, fs.ErrNotExist) {
+		return nil, fmt.Errorf("volume %q already exists (%q)", name, volDir)
+	}
+
+	logrus.Infof("Creating a volume %q", name)
+
+	if err := os.MkdirAll(volDir, 0700); err != nil {
+		return nil, err
+	}
+
+	size, err := cmd.Flags().GetString("size")
+	if err != nil {
+		return nil, err
+	}
+	diskSize, err := units.RAMInBytes(size)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := qemu.CreateDataDisk(volDir, int(diskSize)); err != nil {
+		return nil, err
+	}
+
+	vol, err := store.InspectVolume(name)
+	if err != nil {
+		return nil, err
+	}
+	return vol, nil
+}
+
+func newVolumeListCommand() *cobra.Command {
+	var volumeListCommand = &cobra.Command{
+		Use: "list",
+		Example: `
+To list existing volumes:
+$ limactl volume list
+`,
+		Short:   "List existing Lima volumes",
+		Aliases: []string{"ls"},
+		Args:    cobra.NoArgs,
+		RunE:    volumeListAction,
+	}
+	return volumeListCommand
+}
+
+func volumeListAction(cmd *cobra.Command, args []string) error {
+	allVolumes, err := store.Volumes()
+	if err != nil {
+		return err
+	}
+
+	w := tabwriter.NewWriter(cmd.OutOrStdout(), 4, 8, 4, ' ', 0)
+	fmt.Fprintln(w, "NAME\tSIZE\tDIR\tIN USE BY")
+
+	if len(allVolumes) == 0 {
+		logrus.Warn("No volume found. Run `limactl volume create VOLUME` to create a volume.")
+	}
+
+	for _, volName := range allVolumes {
+		vol, err := store.InspectVolume(volName)
+		if err != nil {
+			logrus.WithError(err).Errorf("volume %q does not exist?", volName)
+			continue
+		}
+		fmt.Fprintf(w, "%s\t%s\t%s\t%s\n", vol.Name, units.BytesSize(float64(vol.Size)), vol.Dir, vol.Instance)
+	}
+
+	return w.Flush()
+}
+
+func newVolumeDeleteCommand() *cobra.Command {
+	var volumeDeleteCommand = &cobra.Command{
+		Use: "delete VOLUME [VOLUME, ...]",
+		Example: `
+To delete a volume:
+$ limactl volume delete VOLUME
+
+To delete multiple volumes:
+$ limactl volume delete VOLUME1 VOLUME2 ...
+`,
+		Aliases: []string{"remove", "rm"},
+		Short:   "Delete one or more Lima volumes",
+		Args:    cobra.MinimumNArgs(1),
+		RunE:    volumeDeleteAction,
+	}
+	return volumeDeleteCommand
+}
+
+func volumeDeleteAction(cmd *cobra.Command, args []string) error {
+	for _, volName := range args {
```

Comment:
I wonder if it makes sense to read the `lima.yaml` files of **all** instances, to see if a volume is still referenced by any stopped instance, and then ask for confirmation before deleting it (with `limactl volume rm -f` overriding the check).

The networking code does a similar thing to shut down managed network daemons when no running instance is using the specific network anymore, so there is precedence for considering all instances.